manifests/golang.yml

```diff
@@ -415,8 +415,6 @@ tests/:
       TestDynamicConfigV1_ServiceTargets: v1.59.0
       TestDynamicConfigV2: v1.59.0
     test_otel_api_interoperability.py: missing_feature
```

Comment:
```suggestion
    test_otel_api_interoperability.py: missing_feature
    test_otel_env_vars.py:
      Test_Otel_Env_Vars: v1.66.0
```

Reply:
i will update this value in this pr once this pr gets merged in:

https://github.com/DataDog/dd-trace-go/pull/2748